internal/config/validate.go

```diff
@@ -9,49 +9,41 @@ func (c *Config) validate() error {
 	var errs InvalidConfigError
 
 	if c.MaxRetries < 0 {
-		errs.appendFieldError("MaxRetries", "was %d, must be greater than or equal to 0", c.MaxRetries)
-	}
-
-	if c.Identifier == "" {
-		errs.appendFieldError("Identifier", "must not be blank")
-	}
-
-	if got, limit := len(c.Identifier), 1024; got > limit {
-		errs.appendFieldError("Identifier", "was %d bytes long, must not be longer than %d", got, limit)
+		errs.appendFieldError("BUILDKITE_SPLITTER_RETRY_COUNT", "was %d, must be greater than or equal to 0", c.MaxRetries)
 	}
 
 	if got, min := c.Parallelism, 1; got < min {
-		errs.appendFieldError("Parallelism", "was %d, must be greater than or equal to %d", got, min)
+		errs.appendFieldError("BUILDKITE_PARALLEL_JOB_COUNT", "was %d, must be greater than or equal to %d", got, min)
 	}
 
 	if got, max := c.Parallelism, 1000; got > max {
-		errs.appendFieldError("Parallelism", "was %d, must not be greater than %d", got, max)
+		errs.appendFieldError("BUILDKITE_PARALLEL_JOB_COUNT", "was %d, must not be greater than %d", got, max)
 	}
 
 	if got, min := c.NodeIndex, 0; got < 0 {
-		errs.appendFieldError("NodeIndex", "was %d, must be greater than or equal to %d", got, min)
+		errs.appendFieldError("BUILDKITE_PARALLEL_JOB", "was %d, must be greater than or equal to %d", got, min)
 	}
 
 	if got, max := c.NodeIndex, c.Parallelism-1; got > max {
-		errs.appendFieldError("NodeIndex", "was %d, must not be greater than %d", got, max)
+		errs.appendFieldError("BUILDKITE_PARALLEL_JOB", "was %d, must not be greater than %d", got, max)
```

Comment:
In your PR description the error reads like this: `BUILDKITE_PARALLEL_JOB was 0, must not be greater than -1`

I don't think this is right?

Reply:
Yeah definitely not right! Will take a look, thanks for noticing!

Reply:
@niceking this is happening because the PARALLEL_JOB_COUNT is set to 0, and it's being dynamically calculated. Are you happy for me to ship this as is and raise another card to improve the configuration reading/validation as suggested by @nprizal below?

Reply:
@niceking This ^^

Reply:
Hmmm just had another read through of this and it seems like the actual validation check is wrong, but the string is right? Like the validation check is: 

```
if got, max := c.NodeIndex, c.Parallelism-1; got > max {
		errs.appendFieldError("BUILDKITE_PARALLEL_JOB", "was %d, must not be greater than %d", got, max)
```

So it's a little confusing that the max is `-1`? 

Anyways I think this needs to go in for v0.9.0, I don't really care if it goes in in this PR or another one, but this should be fixed. Will approve this and you can decide whether you wanna fix the validation in this PR or the next one